### DISCUSSION TITLE
would a ESP8285 chip work with HyperSerialESP8266?

### DISCUSSION THREAD

**Discussion by computergreek** [awawa-dev/HyperSerialEsp8266#26] [Q&A] [answered]
Hi, i recently bought a generic SP511E WiFi LED Controller off of [Amazon](https://www.amazon.com/dp/B09JFW6BSS) and noticed that it doesn't have an 8266 board, it's got an 8285 board. My plan right now is to flash the controller with WLAN and see what wireless performance would be like but I'd also like to explore running it via USB on the host computer running HyperHDR via USB/Serial because that appears to be a better/more popular choice. 

My question is: would HyperSerialEsp8266 still work with the ESP8285 controller linked above or would attempting that brick it?

Thanks so much. love this project

**Comment by awawa-dev** [comment, marked as answer]
Hi
Thanks :) You will probably need to recompile the firmware and change the target platform from esp8266 to esp8285, but this part should be very easy. I don't have it, but the 8285 uses a different clock. It may or may not interfere with the Neopixelbus library clock management for some rendering methods: you can ask in that project if they have encountered esp8285 in practice. You still need a decent serial communication chip on board like the genuine ch340g and you have to be very careful to avoid situations like the esp8285 board in this adapter being powered from the LED strip power supply and the USB port at the same time (they must be separated, only the common ground should be present between these two power supplies).

**Reply by computergreek** [reply]
gotcha. so basically.....it's not raelly worth it.
i ended up buying a [D1 mini](https://www.amazon.com/gp/product/B081PX9YFV/?th=1) and just flashed HyperSerialEsp8266 on it.

here's what I did over the weekend:
![image](https://user-images.githubusercontent.com/38049877/236851418-6667de85-93c6-433e-88ac-be98e9ffa300.png)

I just now re-read what you said in your reply 2 weeks ago and think i now understand it, you mentioned that I needed to avoid having the esp82xx board getting power from the LED strip power supply and the USB port at the same time.
did i wire up the controller wrong?

**Reply by awawa-dev** [reply]
OK, some hints

1) I understand that this is just a test setup, but if you now connect a USB cable to the device that hosts HyperHDR to provide data to HyperSerialEsp8266 on Esp8266, it will also connect a second power source and cause very serious problems. You CANNOT use two power sources for one device (only provide 'common ground' is necessary & required) and by powering the ESP/RaspberryPi via the GPIO chip you circumvent the built-in protections, so you need to know what risks are involved.

2) You shouldn't use an I2C level shifter like in the picture. There is a high risk that it will cause problems because it is too slow for LED data transmission. Check out my latest blog article, and the HyperHDR wiki also mentions the appropriate type of level shifter.

**Reply by awawa-dev** [reply]
3) One power supply for Rpi then connect Esp8266 to Rpi's USB power, second power supply for the LED strip and it's +5V for the level shifter. The common ground  between LED strip and level shifter&Esp8266. Schemes you can also find on the blog.

**Reply by computergreek** [reply]
> 3. One power supply for Rpi then connect Esp8266 to Rpi's USB power, second power supply for the LED strip and it's +5V for the level shifter. The common ground  between LED strip and level shifter&Esp8266. Schemes you can also find on the blog.

😱 i've been doing it wrong this whole time--i had both ground and power connecting to a wire nut to the LED power supply all weekend.

i just disconnected the controller board's red power cable from the wire nut connecting to both ends of the LED light strips. i then left the black ground wire (also coming out from controller board) connected to the wire nut joining the LED light strips. i assume this is how it's all supposed to work?

thankfully, it looks like i somehow avoided serious issues and didn't fry anything.

more on the level shifter in a sec...

**Reply by computergreek** [reply]
> OK, some hints
> 
> 1. I understand that this is just a test setup, but if you now connect a USB cable to the device that hosts HyperHDR to provide data to HyperSerialEsp8266 on Esp8266, it will also connect a second power source and cause very serious problems. You CANNOT use two power sources for one device (only provide 'common ground' is necessary & required) and by powering the ESP/RaspberryPi via the GPIO chip you circumvent the built-in protections, so you need to know what risks are involved.
> 2. You shouldn't use an I2C level shifter like in the picture. There is a high risk that it will cause problems because it is too slow for LED data transmission. Check out my latest blog article, and the HyperHDR wiki also mentions the appropriate type of level shifter.

1. phew, avoided major problems even though i was doing it all wrong.

2. re: level shifter--i can explain. you see, i'm really, really new at this whole thing and honestly have no idea what i'm doing (obviously). i've been struggling with following along with the documentation because i've been lacking familiarity with what everything is, what it does, how it does it, etc. 
I particularly struggled with the logic level shifter and trying to grasp what it was supposed to do conceptually; I now (think?) I understand that it amplifies a 3.3v signal/power to 5v.
I saw that a few years ago, you [suggested ](https://www.hyperhdr.eu/2020/12/my-build-log-using-sk6812-rgbw-led.html#chapter3) that one could use the first LED on the strip to do the conversion but I didn't understand how to actually implement that. I then saw your [updated documentation](https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter2e) saying to use a SN74AHCT125N/74AHCT125 level shifter but coming from someone who hadn't done this sort of thing before, I didn't understand how i was supposed to use it. I looked at the diagram on the [wiki](https://github.com/awawa-dev/HyperHDR/wiki/Level-Shifter) and while the pictures were appreciated, i'm ashamed to say that I still didn't really understand which side was 3.3v vs. 5v and what the logic was with making it all work. 

this specific part of the project isn't very approachable to beginners like me who have never done anything like this before, and it makes sense why--the nature of this part isn't exactly noob-friendly, from a technical perspective.

i needed to watch many hours of youtube videos before feeling comfortable enough to take this on. the most [useful](https://youtu.be/dXLOqGa-n5A) one is the reason why i built the controller board the way i did using that specific I2C level shifter.


i hope this provides you with some context. at the end of all this (assuming I end up getting it right), if time allows I may write up a **very** thorough tutorial for noobs like me to do it right the first time.

**Reply by awawa-dev** [reply]
You can try I2C level shifter but many had problems in the past. The warning is even mentioned in the WLED project help section if you dont believe me ;) https://kno.wled.ge/basics/compatible-hardware/ Esp8266/Esp32/Rpi are all 3.3V logic level and the LED strip data line is 5V.

**Reply by computergreek** [reply]
i definitely believe you.
i'm (so far) pleased to say that i hadn't noticed any problems--what should I expect? I'll see about ordering a handful of the recommended logic level shifters and try to figure out where to go from there.
would i just be able to ...... replace it and keep everything else the same?

**Reply by awawa-dev** [reply]
I tested & recommend only one level shifter SN74AHCT125N, used also by many users and it's working fine. Also sacrificial LED can be used alternatively. Standard I2C level shifters usually work up to around 1Mb speed without any guarantee (I2C doent need such high speeds), sk6812/ws2812 are using 0.8Mb/800Kb speed. Just warned about it.

**Reply by computergreek** [reply]
understood, thank you.
if i wanted to try using a sacrificial LED, how exactly would I go about doing that? what settings would I need to change, and where? what would I need to change on the control board?

Since the LED controller is literally right next to the start of the LED strip, I would want to use the first diode on the strip as the sacrificial one.

**Reply by awawa-dev** [reply]
I dont think it's possible that way..the sacrificial LED needs to have a voltage lowered to ~4V with a diode. That's the point to lower a voltage to make this LED to accept 3.3V data line, you cant use 5V from the main LED strip. You need to separate the sacrificial LED from the rest of the LED strip.

**Reply by computergreek** [reply]
i understand. thank you. i appreciate you taking the time to answer my questions.

so I think i'd like to move forward with your recommendation of using a SN74AHCT125N.
would i just drop it on the board replacing the I2C level shifter like this?
![image](https://user-images.githubusercontent.com/38049877/236901092-ecab8b8c-3447-4162-863b-72c06c7ebda3.png)

**Reply by awawa-dev** [reply]
No, it's impossible to avoid studying the available schematic on a blog or wiki. The wiring is completely different.

**Reply by computergreek** [reply]
yeah that’s fair. i’ve got some reading to do then

**Reply by computergreek** [reply]
> No, it's impossible to avoid studying the available schematic on a blog or wiki. The wiring is completely different.

okay i think i might’ve found a picture that’s pretty noob-friendly from this site: https://hagensieker.com/2021/01/16/logic-level-shifting-and-shit/

does this wiring look about right for what i should aim for?

![](https://hagensieker.com/wp-content/uploads/2021/01/WLED-logic-shifter_bb-1536x898.jpg)

it’s the only one i’ve been able to come across so far that’s easy for me to understand 😬

**Reply by computergreek** [reply]
alright i think i found a candidate from [this site](https://resinchemtech.blogspot.com/2023/01/wiring-diagrams.html)

![image](https://github.com/awawa-dev/HyperSerialEsp8266/assets/38049877/8eea23a2-8db3-4d01-a23e-f05de08eb23a)

so now i feel like i'm good to go unless you tell me that there's something wrong with the examples linked?

**Reply by awawa-dev** [reply]
You are free to use it, although I would not go this way. Recommended wiring schemes created by @pavel-ponomarev are on my blog and on the wiki.

**Reply by computergreek** [reply]
looking back at it, i’m starting to get a better understanding on how it works and what needs to be connected where on a breadboard/circuit

i’ll try making a new one later this weekend after new parts come in

**Reply by computergreek** [reply]
@awawa-dev i bought a handful of [SN74AHCT125N ](https://www.amazon.com/dp/B08R6BCSYC/)level shifters and wired this up based on what i thought @pavel-ponomarev's schematics were supposed to represent:

![image](https://github.com/awawa-dev/HyperSerialEsp8266/assets/38049877/fff7f76f-a5fa-49ab-809e-1968ed2d7fa7)


would you mind confirming whether or not it's right? i don't think i missed anything-- i connected it all up and found that it worked but i wanted to make sure we were good to go before closing it up and forgetting about it

**Reply by awawa-dev** [reply]
Yes, it's fine. You dont need +5V from esp8266 anymore.
